src/support.rs

```diff
@@ -0,0 +1,87 @@
+use crate::palloc::{Inner, InternalAsValue, ToInternal};
+use pgx::*;
+
+// FIXME: This rough translation from C to Rust is _untested_ and probably broken
+
+#[pg_extern(immutable, strict)]
+#[search_path(@extschema@)]
+pub unsafe fn make_call_subquery_support(input: Internal) -> Internal {
+    make_call_subquery_support_inner(input.to_inner()).internal()
+}
+
+pub unsafe fn make_call_subquery_support_inner(
+    input: Option<Inner<*mut pg_sys::Node>>,
+) -> Inner<*mut pg_sys::Node> {
+    let input: *mut pg_sys::Node = input.unwrap().cast();
```

Comment:
If you end up using `input: Inner<pg_sys::Node>` I think this should be
```suggestion
    let input: *mut pg_sys::Node = (&mut *input.unwrap()) as _;
```
or if it's `input: Internal` then
```suggestion
    let input: *mut pg_sys::Node = input.unwrap().unwrap() as _;
```

Reply:
Yes indeed!